src/chainstate/stacks/db/accounts.rs

```diff
@@ -407,8 +380,23 @@ impl StacksChainState {
         Ok(())
     }
 
-    /// Get the scheduled miner rewards in a particular Stacks fork at a particular height
-    pub fn get_scheduled_block_rewards_in_fork<'a>(
+    /// Find the reported poison-microblock data for this block
+    /// Returns None if there are no forks.
+    pub fn get_poison_microblock_report<T: ClarityConnection>(
+        clarity_tx: &mut T,
+        height: u64,
+    ) -> Result<Option<(StacksAddress, u16)>, Error> {
```

Comment:
Is the Poison Microblock report subject to be augmented? Should we introduce a dedicated struct?

Reply:
No, I don't think so.  This code is used to determine whether or not a coinbase recipient should be slashed in favor of some other address that sent a valid `PoisonMicroblock`.  The production code only ever looks at the `StacksAddress` -- the `u16` sequence number is only ever examined by the transaction test code in order to verify that multiple `PoisonMicroblock` reports in the same stream correctly reward the address who reported the earliest such poison.

Reply:
I think the code should continue to report the `u16` sequence number, though, because if the miner wants to include a `PoisonMicroblock` transaction to claim the coinbase from an earlier miner, and someone else has already reported the poison, then this miner will need to see if the poison it detected is earlier in the stream (otherwise there's no point to mining the `PoisonMicroblock` transaction).